### PR TITLE
0a status request problem

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,9 +1,13 @@
 # python-tuya
 
-[![Build Status](https://travis-ci.org/clach04/python-tuya.svg?branch=master)](https://travis-ci.org/clach04/python-tuya)
-
-NOTE I'm no longer using these devices so this is not actively maintained. https://github.com/codetheweb/tuyapi node.js based is active and being updated on a regular basis.
-
+Modified version 
+---------------
+- added support for devices that don't respond to 0a status request command
```
Comment (clach04) [inline README.md:5]: this is more appropriate in commit/PR text.

File: README.md [hunk 2]
```diff
@@ -1,9 +1,13 @@
...
+- updated connection persistence
+- keep reading till footer message
+- used by https://github.com/TradeFace/tuyamqtt
```
Comment (clach04) [inline README.md:8]: nice! a used-by/related section is a good idea

File: README.md [hunk 3]
```diff
@@ -1,9 +1,13 @@
 # python-tuya
 
-[![Build Status](https://travis-ci.org/clach04/python-tuya.svg?branch=master)](https://travis-ci.org/clach04/python-tuya)
```
Comment (clach04) [inline README.md:3]: this needs to stay

File: README.md [hunk 4]
```diff
@@ -1,9 +1,13 @@
...
-
-NOTE I'm no longer using these devices so this is not actively maintained. https://github.com/codetheweb/tuyapi node.js based is active and being updated on a regular basis.
```
Comment (clach04) [inline README.md:5]: sadly this is still true. I'm not using/maintaining the library at the moment.

File: pytuya/__init__.py [hunk 5]
```diff
@@ -19,6 +21,9 @@
 import colorsys
 import binascii
 
+
+from bitstring import BitArray
```
Comment (clach04) [inline pytuya/__init__.py:25]: I don't know what this is

File: pytuya/__init__.py [hunk 6]
```diff
@@ -151,39 +166,87 @@ def __init__(self, dev_id, address, local_key=None, dev_type=None, connection_ti
...
             port (int): The port to connect to.
         """
+
+        self.version = 3.1
         self.id = dev_id
         self.address = address
         self.local_key = local_key
         self.local_key = local_key.encode('latin1')
         self.dev_type = dev_type
-        self.connection_timeout = connection_timeout
```
Comment (clach04) [inline pytuya/__init__.py:159]: this needs to be restored

File: pytuya/__init__.py [hunk 7]
```diff
@@ -268,51 +345,118 @@ def generate_payload(self, command, data=None):
...
-            result = cipher.decrypt(result, False)
-            log.debug('decrypted result=%r', result)
-            if not isinstance(result, str):
-                result = result.decode()
-            result = json.loads(result)
-        else:
-            log.error('Unexpected status() payload=%r', result)
 
-        return result
+    # def availability(self):
```
Comment (clach04) [inline pytuya/__init__.py:451]: commented out code should be removed

File: pytuya/__init__.py [hunk 8]
```diff
@@ -326,13 +470,69 @@ def set_status(self, on, switch=1):
...
+        self.results = []
+        while self.received_footer == False:
+            data = self._send_receive(payload)            
+            self._process_tuya(data)
+            payload = None
+            # print('set reply', bin2hex(data, True), self.received_footer)
+        return self.received_footer
+       
+
+    # def process_status(self, data):
```
Comment (clach04) [inline pytuya/__init__.py:484]: commented out code should be removed

File: pytuya/__init__.py [hunk 9]
```diff
@@ -382,10 +585,6 @@ def set_timer(self, num_secs):
         log.debug('set_timer received data=%r', data)
         return data
 
-class OutletDevice(Device):
```
Comment (clach04) [inline pytuya/__init__.py:385]: this needs to be restored or needs an explanation

**Review by clach04 — CHANGES_REQUESTED** [review]
Thanks for taking the time to send in a PR. As it stands I can't accept this PR as I don't 100% know what its doing. It appears to remove existing functionality in a few places that existing users would be relying on.

I suspect you've improved reliability from some of the changes?

CCing @ziirish who recently also posted a PR and might have some thoughts/opinions - I'm not in a position to run this code against real hardware so I'll need to rely on at least one other person confirming things still work if the backwards-compatibility/existing-functionality is restored.

**Comment by ziirish** [comment]
Sure, I can give it a try early next week though I don't understand all the changes either.

**Comment by ghost** [comment]
Yeah, skip the pr. I rewrote the whole thing here https://github.com/TradeFace/tuya

**Pull request closed without merging**